modules/secure-cloud-function-core/main.tf

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+module "cloudfunction_bucket" {
+  source  = "terraform-google-modules/cloud-storage/google//modules/simple_bucket"
+  version = "~>3.4"
+
+  project_id    = var.project_id
+  labels        = var.labels
+  name          = "gcf-v2-sources-${var.project_number}-${var.location}"
```

Comment:
Use `var.project_id` instead of `var.project_number` so we can require fewer vars

Reply:
can you use the [project data source](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/project) to get the project number from the project_id, instead of using a variable for project number?

Reply:
hey @gtsorbo the issue in using the [project data source](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/project) to get the project number from the project_id is that always trigger a modifying action when re-applying the terraform code.